platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -77,6 +77,22 @@ The extension should only be created by the Gradle Plugin Development plugin. Th
 
 These methods will be removed in Gradle 9.0.
 
+[[validate_plugins_without_stricter_validation]]
+==== Not enabling `ValidatePlugins.enableStricterValidation` is deprecated
+
+It is now required to enable stricter validation of plugin types.
+
+This needs to be done in your local build logic (`buildSrc` or included plugins builds). This is also required if you are simply building Gradle plugins.
+
+[source,kotlin]
+----
+tasks.validatePlugins {
+    enableStricterValidation = true
+}
+----
+
+Starting with Gradle 9.0, stricter validation of plugin types will be enabled by default.
```

Comment:
Seems odd to tell users to enable a flag in 8.14 that they will not need to set in 9.0. Seems like a lot of flip-flopping in build scripts

> Hey set this flag now and you can delete it once you upgrade

I think we should try to avoid solutions which cause users to update all their build scripts, just to have them update them again right after the upgrade. 

Can we instead emit deprecations warnings for all errors that _would_ be thrown if `enableStricterValidation` were set to true? And then turn those deprecations into errors in 9.0? That should prevent users needing to update their build scripts

Reply:
Right. Actually what should be done is to move checks that are currently under "stricter validation" into "regular validation". I'll rework this PR.

Reply:
This turns out more problematic than anticipated. So many plugins are of course not enabling the stricter validation.
Back to the drawing board.

Reply:
We don't use the stricter validation in our plugin validation smoke tests.
The latest published versions of many plugins currently fail when enabling stricter validation, including AGP.

I'm going to close this PR for now as https://github.com/gradle/gradle/issues/22600 should be approached differently. I guess a good first step would be to enable stricter validation in our smoke tests and get prominent plugins in a better state before we can promote the strict rules to regular validation.